### ISSUE TITLE
Tools: more splat editing tools (clone, move etc.)

### ISSUE THREAD

**Issue by aras-p** [aras-p/UnityGaussianSplatting#50]
Sometimes, especially when deleting some unwanted splat areas, you're left with like a "hole" in a surface. In many cases it _feels like_ the hole could be patched up by "simply" copying some nearby splats into it.

So similar to how you can select + delete individual splats right now, maybe you could also select + duplicate + move/rotate individual splats.

And maybe edit their colors (tint, blur, change opacity)? Whelp this sounds like a rabbit hole!

**Comment by unitycoder** [comment]
i was thinking about some tools for cleaning up (but haven't tested yet, will post here if it works)
- Filter by transparency (it would remove almost invisible splats)
- Keep main area only (using camera positions, it would remove splats that are not visible to ALL cameras, basically outside of the "camera path", since often images are taken in circle around main object)

**Comment by aras-p** [comment]
> Filter by transparency

Yeah I was thinking about this one, just did not get around to do it yet.

> it would remove splats that are not visible to ALL cameras

My guess is that literally this would not work. In many captures I've seen quite some cameras are "close" to an "the object" and they don't see whole of it. By this logic, this would cull away parts of said object. That said, maybe something like "remove things not seen by at least 50% of cameras" (or some configurable percentage) would actually work.

But then again, splat files you get from say poly.cam or lumalabs.ai do not even have camera info in them, so a bit of a 🤷

**Comment by unitycoder** [comment]
btw. if i want to loop the points from c# side, do i need to parse them out from gs.asset.m_PosData.bytes (or is there array ready somewhere)? and are they in world or some local pos (in the gs.asset data)?

**Comment by aras-p** [comment]
> if i want to loop the points from c# side, do i need to parse them out from gs.asset.m_PosData.bytes

Their positions are indeed in `asset.m_PosData` array (in local space), laid out in possibly several different formats. In "Very High" quality setting they are just 3 floats per splat position, so that one is easiest. However, that would include all the splats, including the ones that are deleted or cut out.
